flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneJobClusterConfigurationParserFactory.java

```diff
@@ -105,10 +109,14 @@ private int getRestPort(CommandLine commandLine) throws FlinkParseException {
 		}
 	}
 
-	private static JobID getJobId(CommandLine commandLine) throws FlinkParseException {
+	private static JobID getJobId(CommandLine commandLine, Configuration configuration) throws FlinkParseException {
 		String jobId = commandLine.getOptionValue(JOB_ID_OPTION.getOpt());
 		if (jobId == null) {
-			return DEFAULT_JOB_ID;
+			if (configuration.getString(HighAvailabilityOptions.HA_MODE, null) == null) {
```

Comment:
Better to replace with `!HighAvailabilityMode.isHighAvailabilityModeActivated(configuration)`

Reply:
Done